torchqdynamics/ode/forward_solver.py

```diff
@@ -123,17 +124,18 @@ def _adaptive_odeint(self):
 
         # initialize the progress bar
         pbar = tqdm(total=self.t_save[-1].item(), disable=not self.options.verbose)
+        warnings.simplefilter('ignore', TqdmWarning)  # ignore tqdm precision overflow
```

Comment:
The fix looks good to me, but we should use a context manager not to disable warnings globally, i.e. surround the `while` loop with (see [Python doc](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings))
```python
with warnings.catch_warnings():
    warnings.simplefilter('ignore', TqdmWarning)
    # ...
```

Reply:
I'm somewhat sacred of wrapping the whole `while` loop in a context manager (for performance). So I just wrap the `pbar.update` part.

Reply:
Performance-wise the global context manager is better, but it's fine, it's peanuts compared to our matrix multiplications 😋